odl/trafos/wavelet.py

```diff
@@ -750,11 +828,26 @@ def __init__(self, ran, nscales, wbasis, mode):
 
             'per': periodization -- like periodic-padding but gives the
             smallest possible number of decomposition coefficients.
+
+            With J.-O. Stomberg's biorthogonal wavelets only signal
+            extension mode 'sym' is available.
         """
         self.nscales = int(nscales)
-        self.wbasis = wbasis
         self.mode = str(mode).lower()
 
+        if isinstance(wbasis, pywt.Wavelet):
+            self.wbasis = wbasis
+        else:
+            try:
+                wbasis = wbasis + ''
+                if wbasis.startswith('JOS'):
+                    self.wbasis = wbasis
```

Comment:
I.m.o calls to two different libraries should not be in the same file, this makes the imports etc confusing. I think the way it has been done with `XrayTransform` is ok, make a general class and make it call an `...impl` in some other file which contains all the dirty inner workings.